corpus/templates/pages/team.html

```diff
@@ -0,0 +1,399 @@
+                  <a href="{{core.user.executivemember.github}}" target="blank" class="btn btn-lg btn-outline"><i class="fab fa-github fa-2x"></i></a>
+                </div>
+              </div>
+            </div>
+          </div>
+          {% endfor %}
+        </div>
+      </div>
+
+      <div id="CompSoc" role="tabpanel" class="tab-content bg-base-100 border-base-300 rounded-box p-6">
+
+        <h2 class="section-heading">CompSoc Faculty</h2>
+        <br>
+        <div class="flex-container">
+          <!-- Faculty -->
+          {% for fac in faculty %}
+          {% if fac.sig.name == "CompSoc" %}
+          <div class="margins">
+            <div class="card card-bordered w-96 bg-base-300 shadow-xl">
+              <figure class="px-10 pt-10">
+                <img class="profile-pic" {% if fac.profile_picture.url is not None %} src="{{fac.profile_picture.url}}" {% endif %} alt="photo" class="rounded-xl" />
+              </figure>
+              <div class="card-body items-center text-center">
+                <h2 class="card-title">{{fac.user}}</h2>
+                <p class="card-post">{{fac.post}}</p>
+                <br/>
+                <div class="card-actions">
+                  <a href="{{fac.user.executivemember.linkedin}}" target="_blank" class="btn btn-outline btn-lg"><i class="fab fa-2x fa-linkedin"></i></a>
+                </div>
+              </div>
+            </div>
+          </div>
+          {% endif %}
+          {% endfor %}
+        </div>
+
+        <h2 class="section-heading">CompSoc Core</h2>
+        <br>
+        <div class="flex-container">
+
+          <!-- Core -->
+          {% for core in compsoc_core %}
+          <div class="margins">
+            <div class="card card-bordered w-96 bg-base-300 shadow-xl">
+              <figure class="px-10 pt-10">
+                <img class="profile-pic" {% if core.user.executivemember.profile_picture.url is not None %} src="{{core.user.executivemember.profile_picture.url}}" {% endif %} alt="photo" />
+              </figure>
+              <div class="card-body items-center text-center">
+                <h2 class="card-title">{{core.user}}</h2>
+                <h2 class="card-post">{{core.post}}</h2>
+                <br/>
+                <div class="card-actions">
+                  <a href="{{core.user.executivemember.linkedin}}" target="_blank" class="btn btn-outline btn-lg"><i class="fab fa-2x fa-linkedin"></i></a>
+                  <a href="{{core.user.executivemember.github}}" target="_blank" class="btn btn-lg btn-outline"><i class="fab fa-github fa-2x"></i></a>
+                </div>
+              </div>
+            </div>
+          </div>
+          {% endfor %}
+        </div>
+
+        <h2 class="section-heading">Members</h2>
+        <br>
+        <div class="flex-container">
+          <!-- Members -->
+          {% for member in members %}
+          {% if member.sig.name == "CompSoc" %}
```

Comment:
It might be better to do SIG level division in the `views.py` code. Doing it this way will end up with 3 full iterations over the members list. Lots of wasted computation.
Instead, iterate once and segregate members based on SIG in the `views.py` code itself and pass a map to the template with key as SIG and value as a list of members.